File: Daily Challenge/February 2025/Count Number of Bad Pairs.cs
public class Solution {
  public long CountBadPairs(int[] nums) {
    Dictionary<long, long> diffCount = new Dictionary<long, long>();
    long badPairs = 0;

    for (int i = 0; i < nums.Length; ++i) {
      long diff = i - nums[i];
      if (diffCount.ContainsKey(diff)) {
        badPairs += i - diffCount[diff];
        diffCount[diff]++;
      } else {
        badPairs += i;
        diffCount[diff] = 1;
      }
    }

    return badPairs;
  }
}

File: Daily Challenge/February 2025/Count Number of Bad Pairs.cpp
class Solution {
  public:
    long long countBadPairs(vector<int>& nums) {
      map<long long, long long> values;
      long long ans = 0;

      for (int i = 0; i < nums.size(); ++i) {
        long long count = values[i - nums[i]];

        ans += i - count;
        values[i - nums[i]]++;
      }

      return ans;
    }
  };